### ISSUE TITLE
[feature request] Add commenting feature

### ISSUE THREAD

**Issue by anddam** [sobitcorp/metapad#11]
Please add a comment feature commenting current selection or line (if no selection present), just a string (to be quickly set in a field in toolbar) to be added at start of each line.

**Comment by sobitcorp** [comment]
This kind of goes beyond the scope of metapad and raises difficult questions like where would such comments be stored (there's really no good portable way to store comments like this with plain text files).
Does anyone else want such a feature?

**Comment by Gitoffthelawn** [comment]
> This kind of goes beyond the scope of metapad and raises difficult questions like where would such comments be stored (there's really no good portable way to store comments like this with plain text files). Does anyone else want such a feature?

I think it's a very interesting idea, and I've frequently wanted that sort of functionality.  That said, as you mentioned, there really isn't a good portable way to store comments like this with plain text files.  A couple ways I can think of are to use an existing standard (e.g. RTF or Markdown) or to create your own simple proprietary standard that inserts a magic control character sequence to indicate that the next block is a comment.  I think the latter would be easy to implement, but it would certainly expand metapad's scope.  That wouldn't necessarily be a bad thing, because it could easily be toggled via an option, and would provide a truly unique feature that would set metapad apart from virtually all other text editors.

**Comment by sobitcorp** [comment]
Indeed, looks like I misunderstood the OP's request.

In that case, this feature already exists - you're looking for `Edit > Block > Quote (Alt+Q)`. The quote string can be set in `Options > Settings > General > Quote With` and it even supports escape sequences like in Find/Replace.

As for "popup/overlay?" kind of comments, I'll keep it in mind but not going to prioritize it for now. If anyone else wants this feature, please comment so I know.

**Comment by anddam** [comment]
Yes I meant like the current `Quote` feature, that I wasn't aware of.
This is pretty much what I asked except the `Quote With` text field should be in the toolbar for ease of access.

This feature should be also extended to work as a toggle, i.e. this should unquote a quoted line, rather than keep adding quoting symbols as it's doing now.

 

As a separate note I have an character visualization issue with the default value, in the setting panel the default quote character the character is properly displayed, in the file content it's a "missing glyph" block as in the attached image.

![Screenshot_20230107_181220](https://user-images.githubusercontent.com/689872/211162672-b65ad793-6ead-4ddd-b93d-0cbb1eb8faa8.png)

Am I missing some Windows font?

**Comment by sobitcorp** [comment]
> This feature should be also extended to work as a toggle, i.e. this should unquote a quoted line, rather than keep adding quoting symbols as it's doing now.

The problem with making it a toggle is that for some use cases you want to be able to nest quotes (e.g. for email replies, what I think this feature was originally meant for).
That said, adding an Unquote function seems like a good idea. Currently, this can be achieved with `Edit > Block > Strip first character (Alt+Z)` but may require multiple invocations depending on your quote string.

> As a separate note I have an character visualization issue with the default value, in the setting panel the default quote character the character is properly displayed, in the file content it's a "missing glyph" block as in the attached image.

The default quote string is `> ` (right angle bracket and space). For you, looks like it got interpreted as a single unicode character U+203E. This is a bug. What is the value of your `HKEY_CURRENT_USER\Software\metapad\szQuote` in the registry? If it's not "`> `", please try deleting it and see what value is restored.

> That was correct, but now I am curious about what @sobitcorp actually understood, I cannot figure out that by reading the answer.

I was thinking of MS Excel-like comments:
![untitled](https://user-images.githubusercontent.com/58403851/211169748-7e41c640-741e-4c32-8d4c-14e33efa7d8f.PNG)

**Comment by anddam** [comment]
>The problem with making it a toggle is that for some use cases you want to be able to nest quotes (e.g. for email replies, what I think this feature was originally meant for).
>That said, adding an Unquote function seems like a good idea. Currently, this can be achieved with Edit > Block > Strip first character (Alt+Z) but may require multiple invocations depending on your quote string.

This should be then articulated into three features, *quote*, *unquote*, *toggle quote*.

>The default quote string is >  (right angle bracket and space). For you, looks like it got interpreted as a single unicode character U+203E. This is a bug. What is the value of your HKEY_CURRENT_USER\Software\metapad\szQuote in the registry? If it's not "> ", please try deleting it and see what value is restored.

Well it's now `#`, maybe I was getting that result because my default codepage is 65001 (UTF-8, no BOM).


>I was thinking of MS Excel-like comments:

Oh, some like an out of band kind of content.

**Comment by sobitcorp** [comment]
> Well it's now `#`, maybe I was getting that result because my default codepage is 65001 (UTF-8, no BOM).

The default codepage should have no effect on loading/saving registry values. Please tell me what OS and version of metapad (first line of `Help > About`) you're using, and what versions of metapad you have used in the past on the same system (if any). This looks like a potentially serious bug, and I can't seem to reproduce it. Thanks!

**Comment by anddam** [comment]
Sorry for the delay.

I am on Windows 11 Version 22H2 (OS Build 22621.1105), according to `winver`. Metapad is 3.71, I had the 3.6 from @alexd 's [webpage](https://liquidninja.com/metapad/download.html).

`Get-UICulture` says 

    > Get-UICulture
    
    LCID             Name             DisplayName
    ----             ----             -----------
    1033             en-US            English (United States)
